proton-j/src/main/java/org/apache/qpid/proton/message/impl/MessageImpl.java

```diff
@@ -557,6 +604,76 @@ public void setApplicationProperties(ApplicationProperties applicationProperties
     public void setBody(Section body)
     {
         _body = body;
+        _bodySections = null;
+    }
+
+    @Override
+    public Message setBodySections(Collection<Section> sections)
+    {
+        clear();
+        if (sections != null && !sections.isEmpty())
+        {
+            _bodySections = new ArrayList<>(sections);
```

Comment:
For consistency I wondered if this should maybe set _body if there is only 1 element in collection.

Related to earlier comments, should validate or restrict the types allowed.